tests/ui/perfherder/graphs_view_test.jsx

```diff
@@ -265,6 +265,43 @@ test('InputFilter from TestDataModal can filter by application name', async () =
   expect(selectedTests.children[0].text).toBe(fullTestName);
 });
 
+test('Changing the platform dropdown while filtered by text in the Test Data Modal displays expected tests', async () => {
+  const { getByText, getByPlaceholderText, getByTitle } = graphsViewControls();
+
+  fireEvent.click(getByText('Add test data'));
+
+  const platform = getByTitle('Platform');
+  fireEvent.click(platform);
+
+  const windowsPlatform = await waitForElement(() => getByText('windows7-32'));
+  fireEvent.click(windowsPlatform);
+
+  const textInput = await waitForElement(() =>
+    getByPlaceholderText(filterText.inputPlaceholder),
+  );
+  setFilterText(textInput, seriesData2[0].name);
+
+  const existingTest = await waitForElement(() =>
+    getByTitle(seriesData2[0].name),
+  );
+
+  // 'mozilla-central windows7-32 a11yr opt e10s stylo'
+  expect(existingTest).toBeInTheDocument();
+  expect(mockShowModal.mock.calls).toHaveLength(6);
```

Comment:
I am not sure if this length of `6` is correct. @airimovici ?

Reply:
It shouldn't be 6 there, I think this is because it wasn't cleared in another test before it and the value it is kept. 
`mockShowModal.mockClear();` should be used to clear this mock